auth/providers/azure/options.go

```diff
@@ -105,6 +111,17 @@ func (o *Options) Validate() []error {
 			errs = append(errs, errors.New("azure.skip-group-membership-resolution cannot be false when passthrough azure.auth-mode is used"))
 		}
 	}
+
+	if o.AuthMode == ARCAuthMode {
+		if o.ResourceId == "" {
+			errs = append(errs, errors.New("azure.resource-id must be non-empty for authentication using arc mode"))
+		}
+
```

Comment:
Should we ensure following in arc mode:
`SkipGroupMembershipResolution` is never true
`ResolveGroupMembershipOnlyOnOverageClaim` is always true.